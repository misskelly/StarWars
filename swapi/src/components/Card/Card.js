import React from 'react'

const personCard = ({name, homeworld, species, popOfHome}) => {
return (
  <article className='card'>
    <h3>{name}</h3>
    <ul>
      <li>species: {species}</li>
      <li>homeworld: {homeworld}</li>
      <li>population: {popOfHome}</li>
    </ul>
    <button>Favorite</button>
  </article>
)
}

export default personCard;

// Name
// Homeworld
// Species
// Population of Homeworld
// A button to“ Favorite” the person