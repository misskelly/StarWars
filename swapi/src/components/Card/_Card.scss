.card {
  display: inline-block;
  transition: all 0.3s ease;
  border-radius: 5px;
  border: 1px solid hsl(20, 3%, 54%);
  box-shadow: 3px 2px 2px hsl(200, 100%, 1%);
  background: hsla(221, 96%, 48%, 0.351);
  min-width: 250px;
  position: relative;
  padding: 1rem;
  color: hsl(0, 0%, 84%);
}

.card:hover {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
  box-shadow: 5px 5px 5px hsl(204, 25%, 20%);
}

.card-heading {
  padding: 0 0 1rem 0;
  margin-bottom: 1rem;
  text-align: center;
  font-size: 1.5rem;
  font-family: $alien;
  border-bottom: 1px solid $yellow;
}

.fav-btn {
  float: right;
  position: relative;
  margin: 1rem 0 0 0;
  cursor: pointer;

  &:hover {
    background-color: transparent;
    color: $yellow;
  }
}