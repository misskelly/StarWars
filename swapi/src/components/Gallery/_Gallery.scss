.gallery {
  width: 100%;
}

.card-container {
  position: relative;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 50px;
  margin-top: 25px;
  padding: 20px;
  width: 100%;
}

.gallery-heading {
  padding: 3rem 3rem 3rem 8rem;
  text-align: center;
  font-family: $jedi;
  text-transform: uppercase;
  color: hsl(0, 0%, 0%);
  text-shadow: $outline;
  font-size: 5rem;
  letter-spacing: 5rem;
}