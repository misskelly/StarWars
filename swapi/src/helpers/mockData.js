export const mockCleanFilm = {
     scroll: "Turmoil has engulfed the\r\nGalactic Republic. The taxation\r\nof trade routes to outlying star\r\nsystems is in dispute.\r\n\r\nHoping to resolve the matter\r\nwith a blockade of deadly\r\nbattleships, the greedy …",
     title: "The Phantom Menace",
     year: "1999"
   }
export const mockDirtyFilm = {
     opening_crawl: "Turmoil has engulfed the\r\nGalactic Republic. The taxation\r\nof trade routes to outlying star\r\nsystems is in dispute.\r\n\r\nHoping to resolve the matter\r\nwith a blockade of deadly\r\nbattleships, the greedy …",
     title: "The Phantom Menace",
     release_date: "1999-07-03"
   }


