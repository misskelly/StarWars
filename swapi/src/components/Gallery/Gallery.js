import React from 'react'
import Card from '../Card/Card'

const Gallery = ({collection, category}) => {
  console.log('gallery people', collection)
  const getCards = (collectionData) => {
    return collectionData.map((item, index) => {
        return <Card key={`${index}_${item.name}`} item={item} category={category}/>
    })
  }
  const cards = getCards(collection);

    return(
      <section className='Gallery'>
        <h2 className='gallery-heading'>{category}</h2>
      {collection.length > 0 && cards}
        
      </section>
    )
  }

  export default Gallery;