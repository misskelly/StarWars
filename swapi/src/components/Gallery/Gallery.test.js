import React from 'react';
import Gallery from './Gallery';
import { shallow } from 'enzyme';
import Card from '../Card/Card';
import { mockCleanPeople } from '../../helpers/mockData.js';


describe('Gallery', () => {
  let wrapper;
  let activeComponent;
  
  beforeEach(() => {
    activeComponent = 'people';

    wrapper = shallow(<Gallery collection={mockCleanPeople} category={ activeComponent }/>);
  });
  
  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
  
});