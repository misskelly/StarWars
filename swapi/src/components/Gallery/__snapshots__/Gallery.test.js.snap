// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery matches the snapshot 1`] = `
<section
  className="gallery"
>
  <h2
    className="gallery-heading"
  >
    people
  </h2>
  <div
    className="card-container"
  >
    <Card
      category="people"
      item={
        Object {
          "homeworld": "Tatooine",
          "name": "Luke Skywalker",
          "population": "200000",
          "species": "Human",
        }
      }
      key="0_Luke Skywalker"
    />
    <Card
      category="people"
      item={
        Object {
          "homeworld": "Tatooine",
          "name": "C-3PO",
          "population": "200000",
          "species": "Droid",
        }
      }
      key="1_C-3PO"
    />
    <Card
      category="people"
      item={
        Object {
          "homeworld": "Naboo",
          "name": "R2-D2",
          "population": "4500000000",
          "species": "Droid",
        }
      }
      key="2_R2-D2"
    />
    <Card
      category="people"
      item={
        Object {
          "homeworld": "Tatooine",
          "name": "Darth Vader",
          "population": "200000",
          "species": "Human",
        }
      }
      key="3_Darth Vader"
    />
    <Card
      category="people"
      item={
        Object {
          "homeworld": "Alderaan",
          "name": "Leia Organa",
          "population": "2000000000",
          "species": "Human",
        }
      }
      key="4_Leia Organa"
    />
    <Card
      category="people"
      item={
        Object {
          "homeworld": "Tatooine",
          "name": "Owen Lars",
          "population": "200000",
          "species": "Human",
        }
      }
      key="5_Owen Lars"
    />
    <Card
      category="people"
      item={
        Object {
          "homeworld": "Tatooine",
          "name": "Beru Whitesun lars",
          "population": "200000",
          "species": "Human",
        }
      }
      key="6_Beru Whitesun lars"
    />
    <Card
      category="people"
      item={
        Object {
          "homeworld": "Tatooine",
          "name": "R5-D4",
          "population": "200000",
          "species": "Droid",
        }
      }
      key="7_R5-D4"
    />
    <Card
      category="people"
      item={
        Object {
          "homeworld": "Tatooine",
          "name": "Biggs Darklighter",
          "population": "200000",
          "species": "Human",
        }
      }
      key="8_Biggs Darklighter"
    />
    <Card
      category="people"
      item={
        Object {
          "homeworld": "Stewjon",
          "name": "Obi-Wan Kenobi",
          "population": "unknown",
          "species": "Human",
        }
      }
      key="9_Obi-Wan Kenobi"
    />
  </div>
</section>
`;
